source/features/edit-readme.tsx

```diff
@@ -12,13 +13,14 @@ async function init(): Promise<void | false> {
 		return false;
 	}
 
+	const defaultBranch = await getDefaultBranch();
 	const isPermalink = /Tag|Tree/.test(select('.branch-select-menu i')!.textContent!);
-
 	const filename = readmeHeader.textContent!.trim();
-	const pathnameParts = select<HTMLAnchorElement>(`.files [title="${filename}"]`)!.pathname.split('/');
+	const {pathname} = select<HTMLAnchorElement>(`.files [title="${filename}"]`)!;
+	const pathnameParts = parseRoute(pathname);
 	pathnameParts[3] = 'edit'; // Replaces /blob/
 	if (isPermalink) {
-		pathnameParts[4] = await getDefaultBranch(); // Replaces /${tag|commit}/
+		pathnameParts[4] = defaultBranch; // Replaces /${tag|commit}/
```

Comment:
I understand that this matches `edit-files-faster`, but awaiting it here means that we only need to await it if `isPermalink`
```suggestion
		pathnameParts[4] = await getDefaultBranch(); // Replaces /${tag|commit}/
```